app/models/stack.rb

```diff
@@ -73,6 +73,9 @@ def update_deployed_revision(sha)
     recorded_last_deployed_commit = last_deployed_commit
     return if recorded_last_deployed_commit.id == actual_deployed_commit.id
 
+    last_deploy = deploys_and_rollbacks.order(created_at: :desc).first
+    return if !last_deploy.success? && last_deploy.commit_range.include?(actual_deployed_commit)
```

Comment:
You don't need to check the range. Just `last_deploy.until_commit == actual_deployed_commit`